lib/wibox/widget/imagebox.lua

```diff
@@ -6,29 +6,43 @@
 -- @widgetmod wibox.widget.imagebox
 ---------------------------------------------------------------------------
 
+local lgi = require("lgi")
+local Rsvg = lgi.Rsvg
+local cairo = lgi.cairo
+
 local base = require("wibox.widget.base")
 local surface = require("gears.surface")
 local gtable = require("gears.table")
+local gdebug = require("gears.debug")
 local setmetatable = setmetatable
 local type = type
-local print = print
+local math = math
+local string = string
+
 local unpack = unpack or table.unpack -- luacheck: globals unpack (compatibility with Lua 5.1)
 
 local imagebox = { mt = {} }
 
+---Load rsvg handle form image file
+---@tparam string file Path to svg file.
+---@return Rsvg handle
+local function load_svg_and_handle_errors(file)
+    local handle, err = Rsvg.Handle.new_from_file(file)
```

Comment:
`Rsvg` can be `nil`. Either because it isn't compiled with GObject Introspection or because it isn't installed. This is the case by default in many server and KDE based distributions. I will take care of notifying the distributions if this gets merged about added librsvg as a suggested dependency.

Reply:
Rsvg cannot be `nil`, because lgi will throw an error instead of returning `nil`:
```lua
$ lua -e 'lgi = require("lgi") ; a = lgi.does_not_exist ; print("I survived")'
lua: /usr/share/lua/5.2/lgi/namespace.lua:158: Typelib file for namespace 'does_not_exist' (any version) not found
stack traceback:
	[C]: in function 'assert'
	/usr/share/lua/5.2/lgi/namespace.lua:158: in function </usr/share/lua/5.2/lgi/namespace.lua:155>
	(...tail calls...)
	(command line):1: in main chunk
	[C]: in ?
```
(Not much time, I didn't look at anything here, I just saw this comment)

Reply:
Excuse me, I'm not sure how to properly avoid such error throwing. May I use `pcall`?

Reply:
Yup, I guess that's the best approach. Quick attempt (with a typo in `Rsvgg` to show that it is working):
```lua
local lgi = require("lgi")
local Rsvg = nil
do
	-- Load Rsvg if it is available, else set it to nil.
	local success, err = pcall(function()
		Rsvg = lgi.Rsvgg
	end)
	if not success then
		-- I guess this should go to gears.debug.print_warning...?
		print("Could not load Rsvg: " .. tostring(err))
	end
end
print(Rsvg)
```